azure-pipelines.yml

```diff
@@ -0,0 +1,72 @@
+strategy:
+  matrix:
+    UnitTest:
+      TESTKIND: "unit"
+    IntegrationTest:
+      TESTKIND: "integration"
+
+pool:
+  vmImage: 'vs2017-win2016'
+
+variables:
+  GOPATH: '$(system.defaultWorkingDirectory)\gopath' # Go workspace path
+  modulePath: '$(GOPATH)\src\github.com\$(build.repository.name)' # Path to the module's code
+  TERRAFORMVERSION: 0.11.13
+  LOGPARSERVERSION: v0.13.8
+
+steps:
+- pwsh: |
+    New-Item -ItemType "directory" -Path $(GOPATH)\pkg
+    New-Item -ItemType "directory" -Path $(modulePath)
+    Get-ChildItem -Path $(system.defaultWorkingDirectory) -Exclude gopath | Move-Item -Destination $(modulePath)
+    New-Item -ItemType "directory" -Path $(modulePath)\logs
+  displayName: 'Set up the Go workspace'
+
+- script: choco install dep
+  displayName: 'Install dep'
+
+- script: choco install terraform --version $(TERRAFORMVERSION)
+  displayName: 'Install Terraform'
+
+- script: dep ensure
+  workingDirectory: '$(modulePath)'
+  displayName: 'Run `dep ensure`'
+
+- pwsh: |
+    if ("$(TESTKIND)" -eq "unit") {
+      go test -v -parallel 1 $(go list ./... | where-object { $_ -notlike "*/test" }) | Tee-Object -FilePath 'logs/unit.log'
+    }
+
+    if ("$(TESTKIND)" -eq "integration") {
+      go test -v $(go list ./... | where-object { $_ -like "*/test" }) | Tee-Object -FilePath 'logs/integration.log'
+    }
+  workingDirectory: '$(modulePath)'
+  displayName: 'Run tests'
+  continueOnError: true
```

Comment:
Why?

Reply:
We need this here so that the task that publishes the xml test results will run. The entire build pipeline is still marked as failed so it doesn't change much

Reply:
Roger